### PR TITLE
reTerminal - Kernel 5.15

### PR DESCRIPTION
**Pull request by matsujirushi**

### CODE REVIEW

**Comment by lakshanthad** [comment]
Hello @matsujirushi,

Thank you for your PR. Please wait. I will test on my side and merge the PR.

Best Regards,
Lakshantha

File: modules/mipi_dsi/mipi_dsi_drv.c [hunk 1]
```diff
@@ -128,7 +128,8 @@ static struct mipi_dsi_device *mipi_dsi_device(struct device *dev)
 	of_node_put(dsi_host_node);
 	if (!host) {
 		dev_err(dev, "Can't find mipi_dsi_host!");
-		goto error;
+		of_node_put(endpoint);
```
Comment (bigbearishappy) [inline modules/mipi_dsi/mipi_dsi_drv.c:131]: It seems no need to change this code here.
The old code did the same work as the new one.

File: modules/mipi_dsi/panel-ili9881d.c [hunk 2]
```diff
@@ -35,12 +38,15 @@ static int ili9881d_get_modes(struct drm_panel *panel, struct drm_connector *con
 			drm_mode_vrefresh(mode));
 		return -ENOMEM;
 	}
-	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
+
 	drm_mode_set_name(mode);
 
+	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
```
Comment (bigbearishappy) [inline modules/mipi_dsi/panel-ili9881d.c:44]: I don't understand why we need to change the code execution order here.
Exchange the order of drm_mode_set_name and mode-type doesn't make any effect on the code running.
And I can find the same code in latest kernel which the drm_mode_probed_add is execution after the mode->height_mm.

File: modules/mipi_dsi/mipi_dsi_drv.c (hunk 1 above)
Comment (matsujirushi) [reply modules/mipi_dsi/mipi_dsi_drv.c:131]: Thanks for review.
Reverted.

File: modules/mipi_dsi/panel-ili9881d.c (hunk 2 above)
Comment (matsujirushi) [reply modules/mipi_dsi/panel-ili9881d.c:44]: Thanks for review.
Reverted.

**Comment by matsujirushi** [comment]
@lakshanthad 

It looks like rerunning mipi_dsi_probe.

```
[    8.904185] [DSI]mipi_dsi_probe: <---
[    9.038140] mipi_dsi fe700000.dsi.0: failed to attach dsi to host: -517
[    9.039150] [DSI]mipi_dsi_probe: <---
```

**Comment by lakshanthad** [comment]
Hello @matsujirushi,

Sorry for the delay. We were on Labor Day holiday. I will test it again tomorrow and merge the PR.

Thank you.

Best Regards,
Lakshantha

**Comment by bigbearishappy** [comment]
> @lakshanthad
> 
> It looks like rerunning mipi_dsi_probe.
> 
> ```
> [    8.904185] [DSI]mipi_dsi_probe: <---
> [    9.038140] mipi_dsi fe700000.dsi.0: failed to attach dsi to host: -517
> [    9.039150] [DSI]mipi_dsi_probe: <---
> ```

The kernel error code -517 means the driver of mipi_dsi need to be remounted a few moments later. So it's ok to see the 
log "mipi_dsi_probe:" twice here.

**Comment by lakshanthad** [comment]
Hello @matsujirushi @bigbearishappy,

I have tested again on the reTerminal and noticed the following:

1. It works well with Buster
2. However, with Bullseye, there is a rotation problem with auto-rotate

Test 1 (with Bullseye): 

1. As soon as overlays are installed the screen looks like this:

<div align=center></div>

2. After following [these instructions](https://wiki.seeedstudio.com/reTerminal-FAQ/#q11-the-screen-orientation-is-incorrect-after-installing-raspberry-pi-os-bullseye) and reboot the device, the device orientation looks normal

<div align=center></div>

3. However, when I lift the device and place the device in landscape orientation, the orientation changes

<div align=center></div>

4. Landscape (flipped) has same issue 

<div align=center></div>

5. Portrait and portrait (flipped) have no issue 

<div align=center></div>

<div align=center></div>


Could this be a problem with the overlays?

Thank you.

Best Regards,
Lakshantha

**Comment by bigbearishappy** [comment]
> Hello @matsujirushi @bigbearishappy,
> 
> I have tested again on the reTerminal and noticed the following:
> 
> 1. It works well with Buster
> 2. However, with Bullseye, there is a rotation problem with auto-rotate
> 
> Test 1 with Bullseye and [these changes](https://wiki.seeedstudio.com/reTerminal-FAQ/#q11-the-screen-orientation-is-incorrect-after-installing-raspberry-pi-os-bullseye):
> 
> 1. As soon as overlays are installed, the screen looks like this:
> 
> 
> 2. After following [these instructions](https://wiki.seeedstudio.com/reTerminal-FAQ/#q11-the-screen-orientation-is-incorrect-after-installing-raspberry-pi-os-bullseye) and reboot the device, the device orientation looks normal
> 
> 
> 3. However, when I lift the device and place the device in landscape orientation, the orientation changes
> 
> 
> 4. Landscape (flipped) has same issue
> 
> 
> 5. Portrait and portrait (flipped) have no issue
> 
> 
> 
> Test 2 with Bullseye without any changes:
> 
> 1. As soon as overlays are installed, the screen looks like this:
> 
> 
> 2. However, even without following [this](https://wiki.seeedstudio.com/reTerminal-FAQ/#q11-the-screen-orientation-is-incorrect-after-installing-raspberry-pi-os-bullseye), the screen behaves the same as 2,3,4,5 in test 1 above.
> 
> In conclusion, no matter whether we follow [this](https://wiki.seeedstudio.com/reTerminal-FAQ/#q11-the-screen-orientation-is-incorrect-after-installing-raspberry-pi-os-bullseye) or not, the screen behavior for auto-rotate is the same.
> 
> Could this be a problem with the overlays?
> 
> Thank you.
> 
> Best Regards, Lakshantha

@lakshanthad 
This is a issue about imu. It has nothing to do with this merge. Please merge this PR and open a new issue.
Thank you~

**Comment by lakshanthad** [comment]
Okay. Thank you.

**Pull request merged**